### ISSUE TITLE
Example snippets and support for python custom operator

### ISSUE THREAD

**Issue by bearsroom** [NVIDIA/DALI#296]
In [dali/python/nvidia/dali/ops.py](dali/python/nvidia/dali/ops.py) I found a custom python operator example `TFRecordReader` which needs to `GetSchema` from a predefined operator written in C++ and register with `DALI_SCHEMA` and `DALI_REGISTER_OPERATOR`. So I wonder if custom operators purely defined in python will be supported in the future, i.e. embedding python code in C++ execution. That would be really helpful for fast prototyping while keeping the data loading procedure efficient. Thank you!

**Comment by JanuszL** [comment]
Hi,
As you already noticed this is just custom python wrapping code for the Operator defined on the native side.
Answering your question about support for the pure python operators we have such idea on our long-term roadmap but we haven't explored yet how feasible it is and what kind of limitations it will pose.
In general switching between C++ and the Python code could introduce noticeable overhead killing overall DALI performance.
So to sum up, we have such plan but not in the near future.
Tracked as DALI-361.
Br,
Janusz

**Comment by JanuszL** [comment]
It is done in https://github.com/NVIDIA/DALI/pull/732 and will be available in the next release.
